### DISCUSSION TITLE
2.4.10 ISO install fails

### DISCUSSION THREAD

**Discussion by CharlieMK** [Security-Onion-Solutions/securityonion#11109] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

tap

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I did a clean install twice. Installation fails with error (in /root/errors.log):
```
### Cleaning up temp files in /nsm/elastic-agent-workspace[ERROR   ] Command '/opt/so/__salt.tmp.dub9stbi' failed with return code: 1
[ERROR   ] stdout:
[ERROR   ] stderr:
[ERROR   ] retcode: 1
[ERROR   ] {'pid': 206963, 'retcode': 1, 'stdout': '\nInstallation initiated, view install log for further details.', 'stderr': '\n\nInstalling in non-interactive mode.: signal: killed'}
[ERROR   ] Command '/opt/so/__salt.tmp.2j72xfqj' failed with return code: 1
[ERROR   ] stdout:
[ERROR   ] stderr:
[ERROR   ] retcode: 1
[ERROR   ] {'pid': 213158, 'retcode': 1, 'stdout': '\nInstallation initiated, view install log for further details.', 'stderr': '\n\nInstalling in non-interactive mode.: signal: killed'}
      Result: False
```

SO seems to work fine - node and container status is all green. Is this something to worry about? 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by CharlieMK** [comment]
I may have to come back on the fact that SO seems to work fine. The interface seems to work fine, but I'm not sure it is actually processing events. It's a test environment so not entirely sure, but EPS is 0 at the moment. **UPDATE**: syslog ingestion also seems broken.

Similar to (or the same as) https://github.com/Security-Onion-Solutions/securityonion/discussions/11076

**Comment by EddieN17** [comment]
Look at the log in `/opt/so/SO-Elastic-Agent_Installer.log` and see if any more information is there.

Or if you can’t find that log file, try executing the Linux installer again in `/opt/so/saltstack/local/salt/elasticfleet/files/so_agent-installers/` and see what you get. An installation log should be created and maybe that’ll give more information.

**Reply by CharlieMK** [reply]
```
timestamp=2023-08-19T17:33:43.556300489Z level=info message="Installation Progress" Status="Installation Precheck Complete"
timestamp=2023-08-19T17:33:43.556309773Z level=info message="Installation Progress" Status="Extracting Elastic Agent files"
timestamp=2023-08-19T17:33:47.99511145Z level=info message="Installation Progress" Status="Executing Elastic Agent installer"
timestamp=2023-08-19T17:33:47.995151163Z level=info message="Installation Progress" Status="Executing the following: ./so-elastic-agent_source/elastic-agent/elastic-agent install --url=https://securityonion.localtest.com:8220 --enrollment-token=[redacted] --certificate-authorities=/opt/Elastic/SO/soca.crt -n"
timestamp=2023-08-19T17:36:48.028738666Z level=error message="Installation Progress" Context="Installing in non-interactive mode." ErrorDetails="signal: killed"
```

**Reply by EddieN17** [reply]
Given there's a three-minute gap between trying to enroll and the error, my guess is a network problem. Make sure a firewall isn't blocking it or are you using a proxy by any chance?
Try running `curl -f --cacert /etc/pki/ca.crt https://{domain}:8220/api/status` on the manager and see if you get a healthy status. Confirm the fleet server is running by looking at Kibana's elastic fleet agents and `sudo so-status` shows so-elastic-fleet as running

**Reply by CharlieMK** [reply]
- {"name":"fleet-server","status":"HEALTHY"}
- sudo so-status shows that everything is running

Not using a proxy.

**Reply by EddieN17** [reply]
Interesting, I would just try rerunning the elastic agent installer again in the directory I mentioned at the start of this comment. Also, I would try downloading the agent from the web console and see if an endpoint could successfully install the agent (don’t forget to add the ip to the firewall in the web console). Maybe those could help you put the pieces together to find an issue.

**Reply by CharlieMK** [reply]
Rerunning the Linux installer in SO produces a new log but results in the same issue.

**Reply by defensivedepth** [reply]
For future reference - The installers under `/opt/so/saltstack/local/salt/elasticfleet/files/so_agent-installers/` are set to use the `Initial-Endpoints` policy, not the policy needed for the Grid. In other words, it wont work correctly for Grid Nodes.

Running this state on a Grid Node will attempt to install the local Agent:

`sudo salt-call state.apply elasticfleet.install_agent_grid`

**Comment by defensivedepth** [comment]
@CharlieMK 

Are you using virtualization? If so, what kind?

How much memory?

**Reply by CharlieMK** [reply]
Hi @defensivedepth . 

Yes, I am deploying SO on a Proxmox server. SO is given 32 GB memory, 500 GB harddisk, and 4 CPUs.

**Reply by defensivedepth** [reply]
Are the hardisks SSD?

From the logs, it looks like the installer is killing the agent setup because it is taking too long. It should not take 3 min to install.

**Reply by CharlieMK** [reply]
It’s part of a 2TB SSD yes. Should be quite performant, although the cpu is not the fastest (it’s an i5).

**Reply by defensivedepth** [reply]
After setup finishes, what happens if you run the following:

`sudo salt-call state.apply elasticfleet.install_agent_grid`

**Reply by CharlieMK** [reply]
Same error in the logs, killed exactly 3 minutes after starting the installation.

**Reply by defensivedepth** [reply]
What does your InfluxDB dashboards look like? Load average, IO Wait, any Alarms?

**Reply by CharlieMK** [reply]
Looks OK to me: 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63464653/d54da441-a784-47b4-a48e-205f6fc88ea3)

**Reply by CharlieMK** [reply]
Just installed the hotfix, `sudo salt-call state.apply elasticfleet.install_agent_grid` ran successfully now. I'm hoping my other issues (e.g. no alerts) are also solved. I'll keep you posted.

**Reply by CharlieMK** [reply, marked as answer]
yes sir, alerts are coming in, events are coming in as well. So for reference: ran `sudo soup` twice to install https://blog.securityonion.net/2023/08/security-onion-2410-hotfix-20230821-now.html and then `sudo salt-call state.apply elasticfleet.install_agent_grid` and all issues are resolved.
